connect_ext_test.go

```diff
@@ -606,6 +606,30 @@ func TestInvalidHeaderTimeout(t *testing.T) {
 	})
 }
 
+func TestInterceptorReturnsWrongType(t *testing.T) {
```

Comment:
Verifies this section of the client code: https://github.com/bufbuild/connect-go/blob/c9b9bcbabc808ef86e5bedf0d080916c3047df8d/client.go#L103-L106